src/Button/Button.tsx

```diff
@@ -1,16 +1,18 @@
 import React, {forwardRef} from 'react'
 import {ButtonProps} from './types'
 import {ButtonBase} from './ButtonBase'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
 
-const ButtonComponent = forwardRef<HTMLButtonElement, ButtonProps>(
-  ({children, ...props}, forwardedRef): JSX.Element => {
-    return (
-      <ButtonBase ref={forwardedRef} as="button" {...props}>
-        {children}
-      </ButtonBase>
-    )
-  }
-)
+const ButtonComponent: PolymorphicForwardRefComponent<'button', ButtonProps> = forwardRef<
```

Comment:
the fix was to use this helper type

Reply:
It might be nice to audit PRC for files where we aren't using PolymorphicForwardRefComponent (maybe just any file that contains `"forwardRef"` but not `"PolymorphicForwardRefComponent"`?